x/ecocredit/client/utils/util.go

```diff
@@ -0,0 +1,26 @@
+package utils
+
+import (
+	"time"
+
+	sdkerrors "github.com/cosmos/cosmos-sdk/types/errors"
+)
+
+// ParseDate parses a date using the format yyyy-mm-dd.
+func ParseDate(field string, date string) (time.Time, error) {
+	t, err := time.Parse("2006-01-02", date)
+	if err != nil {
+		return t, sdkerrors.ErrInvalidRequest.Wrapf("%s must have format yyyy-mm-dd, but received %v", field, date)
+	}
+	return t, nil
+}
+
+// ParseAndSetDate is as helper function which sets the time do the provided argument if
+// the ParseDate was successful.
+func ParseAndSetDate(dest **time.Time, field string, date string) error {
+	t, err := ParseDate(field, date)
+	if err == nil {
+		*dest = &t
+	}
+	return err
+}
```

Comment:
It might be nice to add this to `types` as it is not necessarily ecocredit client specific. No need to make use of it elsewhere but if we make it available in `types`, we can start using elsewhere. Also, looks like this would close #663 so I've gone ahead and reassigned you and moved to in review.